src/fetchers/git-fetcher.js

```diff
@@ -78,30 +55,47 @@ export default class GitFetcher extends BaseFetcher {
     return path.join(this.dest, constants.TARBALL_FILENAME);
   }
 
-  async fetchFromLocal(override: ?string): Promise<FetchedOverride> {
-    const tarballLegacyMirrorPath = this.getTarballMirrorPath({
+  *getLocalPaths(override: ?string): Generator<?string, void, void> {
+    if (override) {
+      yield path.resolve(this.config.cwd, override);
+    }
+    yield this.getTarballMirrorPath();
+    yield this.getTarballMirrorPath({
       withCommit: false,
     });
-    const tarballModernMirrorPath = this.getTarballMirrorPath();
-    const tarballCachePath = this.getTarballCachePath();
-
-    const tarballMirrorPath =
-      tarballModernMirrorPath && (await fsUtil.exists(tarballModernMirrorPath))
-        ? tarballModernMirrorPath
-        : tarballLegacyMirrorPath && (await fsUtil.exists(tarballLegacyMirrorPath)) ? tarballLegacyMirrorPath : null;
-
-    const tarballPath = override || tarballMirrorPath || tarballCachePath;
+    yield this.getTarballCachePath();
+  }
 
-    if (!tarballPath || !await fsUtil.exists(tarballPath)) {
-      throw new MessageError(this.reporter.lang('tarballNotInNetworkOrCache', this.reference, tarballPath));
+  async fetchFromLocal(override: ?string): Promise<FetchedOverride> {
+    let cachedStream;
+    const triedPaths = [];
+    for (const tarballPath of this.getLocalPaths(override)) {
+      if (tarballPath) {
+        try {
+          cachedStream = await new Promise((resolve, reject) => {
```

Comment:
We may wanna add a comment here explaining the weird `await Promise` thing: `createReadStream` is synchronous but it immediately triggers an `error` event if the file is not found. If nothing handles this event, it crashes node. The problem is we want to try an array of files in order so we need this pattern.